src/config.ml

```diff
@@ -1938,8 +1951,8 @@ let merge_push_reply client (push_config : string) =
     | Dhcp_dns, _ -> true
     | Dhcp_ntp, _ -> true
     | Ifconfig, _ -> true
-    | Route, _ -> true
-    | Route_gateway, _ -> true
+    | Route, _ -> not (Conf_map.mem Route_nopull client)
+    | Route_gateway, _ -> not (Conf_map.mem Route_nopull client)
```

Comment:
I added these, but we need to add more. It's a bit unclear to me what exactly:

    When  used  with  --client  or  --pull, accept options pushed by
    server EXCEPT for routes,  block-outside-dns  and  dhcp  options
     like DNS servers.

Reply:
yes, thanks. I looked into the openvpn code, and the route-nopull disables all "OPT_P_ROUTE | OPT_P_IPWIN32" things, namely:
- client-nat
- route
- route-ipv6
- route-metric
- redirect-gateway
- redirect-private
- block-ipv6
- foreign-option
- ip-win32
- dhcp-option
- tap-sleep
- dhcp-renew
- dhcp-pre-release
- register-dns
- block-outside-dns

(basically, a grep for OPT_P_ROUTE and OPT_P_IPWIN32 in options.c -- done on an old openvpn on my side).

I pushed a commit which completes your commit :)